wallet/udb/testdata/v11.db.go

```diff
@@ -0,0 +1,487 @@
+// revocations are generated from a corresponding ticket as needed.
+//
+// Using different accounts for each combination of
+// {mined,unmined}{ticket,vote,revocation} allows the upgrade test to easily
+// verify that the upgrade and balance functions are performing correctly by
+// checking each account for the expected balance.
+
+
+package main
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"time"
+	"bytes"
+	"math"
+
+	"github.com/decred/dcrwallet/errors"
+	"github.com/decred/dcrd/blockchain/stake"
+	"github.com/decred/dcrd/chaincfg"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/dcrutil"
+	"github.com/decred/dcrd/gcs"
+	"github.com/decred/dcrd/hdkeychain"
+	"github.com/decred/dcrd/txscript"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/dcrwallet/wallet/internal/walletdb"
+	_ "github.com/decred/dcrwallet/wallet/internal/walletdb/bdb"
+	"github.com/decred/dcrwallet/wallet/udb"
+	"github.com/decred/dcrwallet/walletseed"
+)
+
+const dbname = "v11.db"
+
+var (
+	epoch    time.Time
+	pubPass  = []byte("public")
+	privPass = []byte("private")
+)
+
+var chainParams = &chaincfg.TestNet3Params
+
+func main() {
+	err := setup()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "setup: %v\n", err)
+		os.Exit(1)
+	}
+	err = compress()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "compress: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func pay2ssgen(addr dcrutil.Address) []byte {
+	s, err := txscript.PayToSSGen(addr)
+	if err != nil {
+		panic(err)
+	}
+	return s
+}
+
+func pay2ssrtx(addr dcrutil.Address) []byte {
+	s, err := txscript.PayToSSRtx(addr)
+	if err != nil {
+		panic(err)
+	}
+	return s
+}
+
+func pay2sstx(addr dcrutil.Address) []byte {
+	s, err := txscript.PayToSStx(addr)
+	if err != nil {
+		panic(err)
+	}
+	return s
+}
+
+func pay2sstxChange() []byte {
+	//OP_SSTXCHANGE OP_DUP OP_HASH160 0000000000000000000000000000000000000000 OP_EQUALVERIFY OP_CHECKSIG
+	return []byte{0xbd, 0x76, 0xa9, 0x14, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x88, 0xac}
+}
+
+func sstxAddrPush(addr dcrutil.Address, amount int64) []byte {
+	s, err := txscript.GenerateSStxAddrPush(addr, dcrutil.Amount(amount), 0x0058)
+	if err != nil {
+		panic(err)
+	}
+	return s
+}
+
+func dummyTxIn(idx *uint32) *wire.TxIn {
+	var prevHash chainhash.Hash
+	*idx++
+	return wire.NewTxIn(wire.NewOutPoint(&prevHash, *idx, 0), 0, nil)
+}
+
+func stakeBaseTxIn() *wire.TxIn {
+	var prevHash chainhash.Hash
+	return wire.NewTxIn(wire.NewOutPoint(&prevHash, math.MaxUint32, 0), 0, nil)
+}
+
+func ticketSpendTxIn(ticket *wire.MsgTx) *wire.TxIn {
+	th := ticket.TxHash()
+	return wire.NewTxIn(wire.NewOutPoint(&th, 0, 1), 0, nil)
+}
+
+func setup() error {
+	db, err := walletdb.Create("bdb", dbname)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+	seed, err := walletseed.GenerateRandomSeed(hdkeychain.RecommendedSeedLen)
```

Comment:
actually, module dependency graphs aside, it makes sense to just use a predetermined seed here in case we ever want reproducibility when regenerating these test artifacts.

Reply:
Agree on reproducibility bit. My first instinct was to use a fixed seed, but I ended up just following what the other tests were doing...

Fixed [here](https://github.com/decred/dcrwallet/pull/1330/commits/3aad10b6e80991af5195597d2e3e2272b03a961c#diff-a67d73ea53978b7c3212bafb5ecc6ba1R145)